### PR TITLE
enabled wallet restoration from OCP3 agents to OCP4 agents

### PR DESCRIPTION
**Pull request by wadeking98**
removed ownership from dump
added script to create ownership with user on restore

### CODE REVIEW

**Comment by wadeking98** [comment]
@esune @WadeBarnes  please take a look when you have a chance

File: config/backup.conf [hunk 1]
```diff
@@ -43,8 +42,20 @@
...
+postgres=issuer-kit-wallet/agent_inoculation_wallet
+postgres=issuer-kit-wallet/agent_medlab_wallet
+postgres=issuer-kit-wallet/agent_openvp_wallet
+
+
+# Schedule backup for 1am PST; system TZ is PST
+# 0 1 * * * default ./backup.sh -s
+
+# Schedule backup verification for 4am PST; system TZ is PST
+# 0 4 * * * default ./backup.sh -s -v all
```
Comment (esune) [inline config/backup.conf:61]: This file should be reverted to the original, without changes that are deployment specific.

File: openshift/TestConnections.txt [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Uses the 'testConnection' script from https://github.com/BCDevOps/openshift-developer-tools
+# Example:
+# echo -e "backup-postgres"  | xargs -I {} testConnection -n a99fd4-dev -f TestConnections.txt -p {}
+
+google.com:80
+issuer-kit-wallet:5432
```
Comment (esune) [inline openshift/TestConnections.txt:6]: This file should not be in the repository as far as I am concerned, it is deployment specific.

File: openshift/settings.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export PROJECT_NAMESPACE="a99fd4"
+export GIT_URI="https://github.com/BCDevOps/backup-container.git"
+export GIT_REF="master"
```
Comment (esune) [inline openshift/settings.sh:3]: This file should not be committed to the repository.

File: openshift/templates/backup/backup-build.param [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+# Component: .
+# Template File: ../openshift/templates/backup/backup-build.yaml
+#=========================================================
+NAME=backup-postgres
+GIT_REPO_URL=https://github.com/BCDevOps/backup-container.git
+GIT_REF=master
+SOURCE_CONTEXT_DIR=/docker
+DOCKER_FILE_PATH=Dockerfile
+OUTPUT_IMAGE_TAG=latest
+BASE_IMAGE_FOR_BUILD=docker.io/centos/postgresql-12-centos7:20200917-804ef01
```
Comment (esune) [inline openshift/templates/backup/backup-build.param:12]: This file should not be committed to the repository.

File: openshift/templates/backup/backup-deploy.dev.param [hunk 5]
```diff
@@ -0,0 +1,50 @@
+#=========================================================
```
Comment (esune) [inline openshift/templates/backup/backup-deploy.dev.param:1]: This file should not be committed to the repository

File: openshift/templates/backup/backup-deploy.param [hunk 6]
```diff
@@ -0,0 +1,42 @@
+#=========================================================
```
Comment (esune) [inline openshift/templates/backup/backup-deploy.param:1]: This file should not be committed to the repository

File: openshift/templates/backup/backup-deploy.prod.param [hunk 7]
```diff
@@ -0,0 +1,50 @@
+#=========================================================
```
Comment (esune) [inline openshift/templates/backup/backup-deploy.prod.param:1]: This file should not be committed to the repository

File: openshift/templates/backup/backup-deploy.test.param [hunk 8]
```diff
@@ -0,0 +1,50 @@
+#=========================================================
```
Comment (esune) [inline openshift/templates/backup/backup-deploy.test.param:1]: This file should not be committed to the repository

**Review by esune — CHANGES_REQUESTED** [review]
Files that are not part of the general repository should not be committed back. I'd suggest adding `.gitignore.` entries to prevent committing `.param` and `settings.sh` files altogether since this repo is a "template" repo - c.c. @WadeBarnes

File: docker/backup.postgres.plugin [hunk 9]
```diff
@@ -89,6 +90,19 @@ function onRestoreDatabase(){
       gunzip -c "${_fileName}" | psql -v ON_ERROR_STOP=1 -x -h "${_hostname}" ${_portArg} -d "${_database}"
       # Get the status code from psql specifically.  ${?} would only provide the status of the last command, psql in this case.
       _rtnCd=${PIPESTATUS[1]}
+      IFS=\0;
```
Comment (WadeBarnes) [inline docker/backup.postgres.plugin:93]: Is there an easier way to do this?  It feels overly complicated.

**Review by WadeBarnes — CHANGES_REQUESTED** [review]
See comments

File: docker/backup.postgres.plugin [hunk 10]
```diff
@@ -58,6 +58,7 @@ function onRestoreDatabase(){
     _portArg=${_port:+"-p ${_port}"}
     _username=$(getUsername ${_databaseSpec})
     _password=$(getPassword ${_databaseSpec})
+    _schema='public'
```
[the remark this answers is not in the document]
Comment (esune) [reply docker/backup.postgres.plugin:61]: This was something we were wondering about, I am guessing we should add a parameter to specify the alternate schema name - if necessary.
@wadeking98 do you remember what was the potential issue - if there is one - with schemas other than default with this recovery process? Can you fish out and post the link to the online resources we used yesterday so we can check them again?

File: docker/backup.postgres.plugin (hunk 9 above)
Comment (esune) [reply docker/backup.postgres.plugin:93]: The easier/cleaner way would be to switch over to using `pg_restore`. I am a fan of switching over to that, but I thought you had some reservations. Changes would at that point need to include:
- unzipping backup file to a `temp` location
- run `pg_restore` with the appropriate flags
- clean up
I think it is doable, but will require some additional testing as the restore process will likely have a couple of differences.

Comment (WadeBarnes) [reply docker/backup.postgres.plugin:93]: I was wondering, more, if there was an more concise way of restoring the permissions on the database with this current approach.

If we were to introduce `pg_restore` based backup and restore capabilities, I'd want to introduce it as an option.  Likely as an add-on plug-in that would be loaded based on a configuration option, and provide completely alternate implementations for `onBackupDatabase` and `onRestoreDatabase`.

File: docker/backup.postgres.plugin (hunk 10 above)
[the remark this answers is not in the document]
Comment (WadeBarnes) [reply docker/backup.postgres.plugin:61]: Here is a starting point for a search/review; https://github.com/BCDevOps/backup-container/blob/master/docker/backup.settings#L10

**Pull request closed without merging**